hyperspy/axes.py

```diff
@@ -594,17 +594,17 @@ def value2index(self, value, rounding=round):
             if rounding is round:
                 #Use argmin(abs) which will return the closest value
                 # rounding_index = lambda x: np.abs(x).argmin()
-                index = numba_closest_index_round(self.axis,value)
+                index = numba_closest_index_round(self.axis,value).astype(int)
             elif rounding is math.ceil:
                 #Ceiling means finding index of the closest xi with xi - v >= 0
                 #we look for argmin of strictly non-negative part of self.axis-v.
                 #The trick is to replace strictly negative values with +np.inf
-                index = numba_closest_index_ceil(self.axis,value)
+                index = numba_closest_index_ceil(self.axis,value).astype(int)
             elif rounding is math.floor:
                 #flooring means finding index of the closest xi with xi - v <= 0
                 #we look for armgax of strictly non-positive part of self.axis-v.
                 #The trick is to replace strictly positive values with -np.inf
-                index = numba_closest_index_ceil(self.axis,value)
+                index = numba_closest_index_floor(self.axis,value).astype(int)
```

Comment:
Looks Good To Me.

However, this behaviour is weird because in the `numba_closest_index` functions, the returned array is initialized with `"uint"` dtype and argmax / argmin functions should not return anything else than `int` types. I wonder what causes this behaviour... 

Well, if it works like this then who am I to judge?

Reply:
I agree, this is odd, there should be a bug somewhere and from a quick look, it should not return float type and casting explicitly to `int` after calling jitted function shouldn't be necessary...

Reply:
I thought so as well.

However, for some reason a statement such as `s.isig[value2index(x)+1:]` would throw an error for `value2index(x)+1` not being an `int` that can be used as index. It is explicitly for when another integer is added/subtracted to the result of `value2index` It is the source of the failure of the tests in https://github.com/LumiSpy/lumispy/pull/78#issuecomment-855796214
As the failure appeared only recently and only for the `non_uniform_axes` branch it must be a result of rewriting the `value2index` function.

Adding the `.astype(int)` fixes this problem. (Note that the `value2index` function for `UniformDataAxis` also explicitly casts to `int`.)  Therefore, I propose to go for this "workaround".

Reply:
>  Therefore, I propose to go for this "workaround".

I share this opinion